src/protocol/message.rs

```diff
@@ -104,7 +104,48 @@ pub enum Message {
     Reject(Reject),
 }
 
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum MessageKind {
```

Comment:
I'm on the fence about this -- I know its not required, but it allows changing the following `assert` style:

Before:
```
assert!(matches!(message, Message::Block(..)), "Expected Block, got {:?}", message);
```
where the formatting is required to get a reasonable panic message.

After:
```
assert_eq!(message.kind(), MessageKind::Block);
```

Reply:
We still need to match in some instances (to check the message internals). Simple current example:
```
match message {
   Message::Pong(nonce) => assert_eq!(nonce, expected_nonce),
   msg => panic!("Expected Pong, got {:?}, msg),
}
```
So we want something where we can easily destructure into content as well as assert on the basic type.

I'm not sure there's a way to do both with just fields - I think one will always need an additional method.